File: src/main/java/ac/dnd/server/admission/domain/model/EventData.java
package ac.dnd.server.admission.domain.model;

import java.time.LocalDateTime;

import ac.dnd.server.common.support.EventStatus;

public record EventData(
	Long id,
	String name,
	ViewablePeriod period,
	LocalDateTime resultAnnouncementDateTime,
	EventStatus status
) {
	public boolean isCompleted() {
		return status.isPublished();
	}

	public boolean isPending() {
		return status.isPending();
	}

	public boolean isAnnounceable(final LocalDateTime currentTime) {
		return isPending() && resultAnnouncementDateTime.isAfter(currentTime);
	}

}

File: src/main/kotlin/ac/dnd/server/admission/domain/model/EventData.kt
package ac.dnd.server.admission.domain.model

import ac.dnd.server.admission.domain.enums.EventStatus
import java.time.LocalDateTime

data class EventData(
    val id: Long,
    val name: String,
    val period: ViewablePeriod,
    val resultAnnouncementDateTime: LocalDateTime,
    val status: EventStatus
) {
    fun isCompleted(): Boolean {
        return status.isPublished()
    }

    fun isPending(): Boolean {
        return status.isPending()
    }

    fun isAnnounceable(currentTime: LocalDateTime): Boolean {
        return isPending() && resultAnnouncementDateTime.isAfter(currentTime)
    }
}
